tests/frontend-storage/cypress/integration/FrontendStorage.test.ts

```diff
@@ -2,34 +2,71 @@
  * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
  * See LICENSE.md in the project root for license terms and full copyright notice.
  *--------------------------------------------------------------------------------------------*/
-import { FrontendStorage } from "@itwin/object-storage-core";
-import { BackendClient } from "../../src/frontend/BackendClient";
-import { RestClient } from "../../src/frontend/RestClient";
+import { FrontendStorage } from "@itwin/object-storage-core/lib/frontend";
+import {
+  TestCase,
+  ServerStorageProxyFrontend,
+  FrontendTestRemoteDirectoryManager,
+  testDownload,
+  testUpload,
+} from "../../src/frontend";
 
-const serverBaseUrl = (window as any).serverBaseUrl;
-const frontendStorage = (window as any).frontendStorage;
+const serverBaseUrl: string = (window as any).serverBaseUrl;
+const frontendStorage: FrontendStorage = (window as any).frontendStorage;
+const serverStorage = new ServerStorageProxyFrontend(serverBaseUrl);
+const directoryManager = new FrontendTestRemoteDirectoryManager(serverStorage)
 
 describe(`${FrontendStorage.name}: ${frontendStorage.constructor.name}`, () => {
-
-  const backendClient = new BackendClient(serverBaseUrl, new RestClient());
+  const test: TestCase = {serverStorage, directoryManager, frontendStorage};
 
   before(() => {
-    cy.visit(backendClient.entrypoint);
+    cy.visit(`${serverBaseUrl}/index.html`);
+  });
+  beforeEach(async () => {
+    await directoryManager.purgeCreatedDirectories();
   });
-
   after(async () => {
-    await backendClient.cleanup();
+    await directoryManager.purgeCreatedDirectories();
   });
-
-  it("should download file from url", async () => {
-    const testFilePayload = "test payload";
-    const downloadUrl: string = await backendClient.getTestDownloadUrl(testFilePayload);
-    expect(downloadUrl).to.not.be.empty;
-
-    // TODO: add correct assertions
-    // const downloadedFileContents: Buffer = (await frontendStorage.download({
-    //   transferType: "buffer",
-    //   url: downloadUrl
-    // }));
+  
+  describe(`${frontendStorage.download.name}()`, () => {
+    it("should download to buffer using URL", async () => {
+      await testDownload(test, "buffer", "url");
```

Comment:
this is kinda hard to read, I don't think there's enough repetitive code in `testDownload` to warrant moving it out like that. If you want to strictly follow the DRY principle you could just have one `it` inside a loop that iterates over an array of test cases - that's a more common pattern. Each test case could include transfer type and input method.

Reply:
Thanks, loops make the tests way more concise
Should I move everything from test-templates/Download.ts & Upload.ts to the .tests.ts file itself?

Reply:
yeah that would be great